src/widgets/text.lua

```diff
@@ -61,6 +61,35 @@ function Text:layout(width)
     end
 end
 
+--- Draw text substuting in Conky variables.
+-- Text line will be updated on each cycle as per Conky's text
+-- Section, some variables such as formatting and positioning
+-- may not be honored.
+-- @type ConkyText
+local ConkyText = util.class(Text)
+w.ConkyText = ConkyText
+
+--- @string text Text to be displayed, can include conky variables.
+--- @tparam table args table of options, see `Text:init`
+function ConkyText:init(text, args)
+    Text.init(self, args)
+
+    self._lines = {}
+    local _, line_count = text:gsub("[^\n]*", function(line)
+        table.insert(self._lines, line)
+    end)
+    self.height = line_count * self._line_height
+end
+
+function ConkyText:render(cr)
+    ch.set_font(cr, self._font_family, self._font_size, self._font_slant,
+                    self._font_weight)
+    cairo_set_source_rgba(cr, unpack(self._color))
+    for i, line in ipairs(self._lines) do
+        local y = self._baseline_offset + (i - 1) * self._line_height
+        self._write_fn(cr, self._x, y, conky_parse(line))
```

Comment:
A while ago `conky_parse` was very slow, see [this issue](https://github.com/brndnmtthws/conky/issues/967). I don't know if that has since been fixed or if it is still relevant. At the time I implemented the eager loader in data.lua to centralize retrieving data from conky_parse.

Have you tried using this in an example with many instances? It would be interesting to observe if there is any slowdown/lag compared to plain conky text.

Reply:
It seems like that issue has been marked as resolved, so far i've really only played with a `CPU Top` example, which is using 10 calls which seemed fine performance wise, I can make that name change tomorrow.